### PR TITLE
Implement time-series widgets styles

### LINKED ISSUE #786: Add time-series widget (first iteration)
Main issue to keep track of WIP

- [x] Required layout changes for new widget #783
- [x] Placeholder views for the various concepts (using fake model/data); controls (play/pause), step info, histogram chart #787
- [x] Implement default date/time histogram #811
- [x] Histogram: format X-axis labels with time and/or date (needs to adapt to range span+mockups) #843
- [x] Torque: implement animation controls (play/pause, time-marker) #844
- [x] Torque: animation for a selected time-series range #855 + https://github.com/CartoDB/torque/pull/243
- [x] Torque: format time-marker step info (time+date) #855 
- [x] CSS for the various layouts/responsive sizes #879

Windshaft needs to be updated to support date column for histogram data :point_right: https://github.com/CartoDB/Windshaft/issues/405

### PR DESCRIPTION
**Pull request by viddo**
Final waypoint to resolve #786 :tada: 

This implements the missing styles for the time-series widget:

Styleguide elements:
screen shot 2015-11-27 at 21 10 00
screen shot 2015-11-27 at 21 10 04
screen shot 2015-11-27 at 21 10 22

Actual usage:
screen shot 2015-11-27 at 21 10 39
screen shot 2015-11-27 at 21 10 46
screen shot 2015-11-27 at 21 11 04

Moved the duplicated resize behavior to be handled internally in the histogram chart view instead of being duplicated for all usages (see code for details of how it was achieved). Also tried to follow backbone-idioms of how view is updated (i.e. utilize model state/events instead of stateful values directly on the view).

Added a new style guide file to test the `layout-loading` styles :art:

While at it, simplified connect task, only run one server instead of 3 on different ports. Added instructions where to find different things to the grunt output, here copied:
```
Started connect web server on http://localhost:9001
Styleguide:
 - http://0.0.0.0:9001/themes/styleguide

Examples:
 - http://0.0.0.0:9001/examples

Jasmine specs available at (one per bundle):
 - http://0.0.0.0:9001/test/SpecRunner-src.html
 - http://0.0.0.0:9001/test/SpecRunner-torque.html
 - http://0.0.0.0:9001/test/SpecRunner-odyssey.html
 - http://0.0.0.0:9001/test/SpecRunner-cartodb.html
```

@xavijam can you review please?
+ @javierarce for the modifications of the chart internals

### CODE REVIEW

**Comment by javierarce** [comment]
:+1: and :dancer: to the chart modifications

Plus the simplification of the server stuff is :top:

File: src/geo/ui/widgets/histogram/chart.js [hunk 1]
```diff
@@ -21,28 +26,52 @@ module.exports = View.extend({
...
+    this.model = new Model({
+      showLabels: true,
+      data: this.options.data,
+      height: this.options.height,
+      margin: _.clone(this.options.margin),
+      width: 0, // will be set on resize listener
+      pos: { x: 0, y: 0 }
+    });
+
+    this.model.bind('change:width', this._onChangeWidth, this);
```
Comment (xavijam) [inline src/geo/ui/widgets/histogram/chart.js]: woah, lot of bindings!

File: src/geo/ui/widgets/time-series/placeholder.tpl [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<div class="Widget-header Widget-header--timeSeries">
+  <% if (hasTorqueLayer) { %>
+    <div class="Widget-timeSeriesFakeControl"></div>
+    <div class="Widget-timeSeriesTimeInfo Widget-timeSeriesTimeInfo--fake"></div>
+  <% } %>
+</div>
+<div class="Widget-content Widget-content--timeSeries">
+  <div class="Widget-timeSeriesFakeChart <% if (hasTorqueLayer) { %>Widget-timeSeriesFakeChart--torque<% } %>">
+    <% for (var i = 0; i < 50; i++) { %>
+      <div class="Widget-timeSeriesFakeChartItem" style="height: <%- Math.floor(Math.random() * 100) %>%"></div>
```
Comment (xavijam) [inline src/geo/ui/widgets/time-series/placeholder.tpl]: :dancers:

File: src/geo/ui/widgets/time-series/torque-time-marker-view.js [hunk 3]
```diff
@@ -6,21 +7,29 @@ var View = require('cdb/core/view');
...
     if (!this.options.chartView) throw new Error('chartView is required');
-    if (!this.options.viewModel) throw new Error('viewModel is required');
     if (!this.options.torqueLayerModel) throw new Error('torqeLayerModel is required');
 
     this._chartView = this.options.chartView;
-    this._viewModel = this.options.viewModel;
     this._torqueLayerModel = this.options.torqueLayerModel;
+    this.viewModel = new Model();
 
     this._torqueLayerModel.bind('change:step', this._onChangeStep, this);
```
Comment (xavijam) [inline src/geo/ui/widgets/time-series/torque-time-marker-view.js]: I like to move all bindings to a separate method, but :+1:.

File: src/geo/ui/widgets/widget-view-factory.js [hunk 4]
```diff
@@ -8,7 +8,7 @@ var WidgetViewFactory = function(defs) {
   }, this);
 };
 
-WidgetViewFactory.prototype.DEFAULT_CLASS_NAMES = 'Widget Widget--light Widget--time';
```
Comment (xavijam) [inline src/geo/ui/widgets/widget-view-factory.js]: Great, I wanted to ask about what happened with this ```Widget--time class``` :+1:

File: themes/scss/widgets/default.scss [hunk 5]
```diff
@@ -1,4 +1,14 @@
 // Default styles for widgets
+//
+// Expected layout hierarchy:
```
Comment (xavijam) [inline themes/scss/widgets/default.scss]: Thanks for this buddy!

File: themes/scss/widgets/default.scss [hunk 6]
```diff
@@ -26,12 +39,17 @@
   margin-right: $sMargin-section;
   margin-left: $sMargin-section;
 }
+.Widget-header--timeSeries {
```
Comment (xavijam) [inline themes/scss/widgets/default.scss]: Why not adding this kind of "layout" in layout stylesheet?

File: themes/scss/widgets/time-series.scss [hunk 7]
```diff
@@ -0,0 +1,78 @@
+// Time-series widget, i.e. histogram of dates
```
Comment (xavijam) [inline themes/scss/widgets/time-series.scss]: :joy: super! :dancer:

**Comment by xavijam** [comment]
Super dupper work man! :+1:.

File: themes/scss/widgets/default.scss (hunk 6 above)
Comment (viddo) [inline themes/scss/widgets/default.scss]: I wasn't aware of that component, I'll have a look to see how I would go about changing this then.

Comment (viddo) [inline themes/scss/widgets/default.scss]: After looking into it, I don't see that it's feasible. If you look [further down in the same file](https://github.com/CartoDB/cartodb.js/pull/879/files#diff-761c5e7aef3c026da769f41f9e4447dbL49) I have other overrides of the layout that are specific for this context, so I need this modifier either way.

Comment (xavijam) [inline themes/scss/widgets/default.scss]: :+1:

**Pull request merged**